Heber/src/main/java/springApp/java/com/example/demo/controllers/InstrumentoController.java

```diff
@@ -0,0 +1,156 @@
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import springApp.java.com.example.demo.exceptions.InstrumentoNoEncontradoException;
+import springApp.java.com.example.demo.models.AccionModel;
+import springApp.java.com.example.demo.models.BonoModel;
+import springApp.java.com.example.demo.models.InstrumentoFinancieroModel;
+import springApp.java.com.example.demo.services.InstrumentoService;
+import springApp.java.com.example.demo.exceptions.InstrumentoDuplicadoException;
+
+import java.util.List;
+import java.util.Optional;
+
+@RestController
+@RequestMapping("/api/instrumentos") // Este es el path base del controller
+public class InstrumentoController {
+
+    @Autowired
+    InstrumentoService instrumentoService;
+
+    @GetMapping("/hola")
+    public String instrumento() {
+        return "asdasd";
+    }
+
+    // GET URL: localhost:8081/service-service
+    @RequestMapping("/hello-service/{id}")
+    public String helloService(@PathVariable("id") Long id) {
+        System.out.println("el id es: " + id);
+        return instrumentoService.helloService();
+    }
+
+
+    // GET URL: localhost:8081/hello-repository
+    @RequestMapping("/hello-repository")
+    public String helloRepository() {
+        return instrumentoService.helloRepository();
+    }
+
+    ///////////METODOS CRUD/////////////////
+    // AGREGAR INSTRUMENTO
+    //http://localhost:5000/api/instrumentos/agregarInstrumento
+    //EJEMPLO DE JSON PARA AGREGAR ACCION
+    /*
+    {
+        "tipo":"ACCION",
+        "nombre": "BYMA",
+        "precio": 100,
+        "cantidad": 10,
+        "dividendo": 10
+    }
+     */
+
+    //EJEMPLO DE JSON PARA AGREGAR BONO
+    /*
+    {
+        "tipo":"BONO",
+        "nombre": "AL30",
+        "precio": 100,
+        "cantidad": 10,
+        "interes": 10
+    }
+     */
+    @PostMapping("/agregarInstrumento")
+    public ResponseEntity<String> agregarInstrumento(@RequestBody InstrumentoFinancieroModel instrumento) {
+        try {
+            if (instrumento instanceof AccionModel) {
+                instrumentoService.agregarAccion((AccionModel) instrumento);
+                return ResponseEntity.status(HttpStatus.CREATED).body("Accion agregada");
+            } else if (instrumento instanceof BonoModel) {
+                instrumentoService.agregarBono((BonoModel) instrumento);
+                return ResponseEntity.status(HttpStatus.CREATED).body("Bono agregado");
+            } else {
+                return ResponseEntity.status(HttpStatus.BAD_REQUEST).body("Tipo de instrumento no valido");
+            }
+        } catch (InstrumentoDuplicadoException e) {
+            return ResponseEntity.status(HttpStatus.CONFLICT).body(e.getMessage());
+        } catch (Exception e) {
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST).body("Error al agregar instrumento");
+        }
+    }
```

Comment:
Como charlamos, ya que tenes un GlobalExceptionHandler, la idea seria migrar todo a un mismo sitio y dejar el controller sin Try.

Tambien el naming de la URL, en donde la idea es seguir un standard, donde ya tenemos el instrumento como ruta base, no hace falta repetir esto en la defincion del enpoint, tambien se interpreta que es agregar dado que es un metodo post, por lo que la ruta quedaria mas sencilla con solo la "/".

```suggestion
    @PostMapping("/")
    public ResponseEntity<String> agregarInstrumento(@RequestBody InstrumentoFinancieroModel instrumento) {
            if (instrumento instanceof AccionModel) {
                instrumentoService.agregarAccion((AccionModel) instrumento);
                return ResponseEntity.status(HttpStatus.CREATED).body("Accion agregada");
            } else if (instrumento instanceof BonoModel) {
                instrumentoService.agregarBono((BonoModel) instrumento);
                return ResponseEntity.status(HttpStatus.CREATED).body("Bono agregado");
            } else {
                return ResponseEntity.status(HttpStatus.BAD_REQUEST).body("Tipo de instrumento no valido");
            }
    }
```